codec-http2/src/main/java/io/netty/handler/codec/http2/AbstractHttp2StreamChannel.java

```diff
@@ -817,7 +834,11 @@ private void updateLocalWindowIfNeeded() {
             if (flowControlledBytes != 0) {
                 int bytes = flowControlledBytes;
                 flowControlledBytes = 0;
-                write0(parentContext(), new DefaultHttp2WindowUpdateFrame(bytes).stream(stream));
+                write0(parentContext(), new DefaultHttp2WindowUpdateFrame(bytes).stream(stream))
+                        // Add a listener which will notify and teardown the stream
+                        // when a window update fails.
+                        // See https://github.com/netty/netty/issues/9663
+                        .addListener(WINDOW_UPDATE_FRAME_LISTENER);
```

Comment:
Should the ChannelFuture be attached to the child channel?

Reply:
@carl-mastrangelo not sure I follow. The returned `ChannelFuture` belongs to the child channel.

Reply:
What I mean is shouldn't this just teardown the stream channel, not the parent channel?   If there was a failure applying the window for the child, it seems only the child should be torn down.

Reply:
@carl-mastrangelo yes this was what it supposed to do but I messed up 🤦‍♂ ... Good catch.